### PR TITLE
team_planet_express_user_id_in_action

### PR DESCRIPTION
**Pull request by arthurkov98**

### CODE REVIEW

File: team_planet_express/src/main/java/core/services/actions/customer/ListCartItemsService.java [hunk 1]
```diff
@@ -17,29 +16,23 @@ public class ListCartItemsService {
...
-        this.currentUserId = currentUserId;
         this.cartService = new CartService(database);
     }
 
-    public MutableLong getCurrentUserId() {
-        return currentUserId;
-    }
-
     public ListCartItemsResponse execute(ListCartItemsRequest request) {
         List<CoreError> errors = validator.validate(request);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/actions/customer/ListCartItemsService.java:31]: Id in validator?

File: team_planet_express/src/main/java/core/services/actions/customer/RemoveItemFromCartService.java [hunk 2]
```diff
@@ -17,25 +16,20 @@ public class RemoveItemFromCartService {
...
-        this.currentUserId = currentUserId;
         this.validator = validator;
     }
 
-    public MutableLong getCurrentUserId() {
-        return currentUserId;
-    }
 
     public RemoveItemFromCartResponse execute(RemoveItemFromCartRequest request) {
         List<CoreError> errors = validator.validate(request);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/actions/customer/RemoveItemFromCartService.java:27]: Id in validator?

File: team_planet_express/src/main/java/core/services/actions/guest/SignUpService.java [hunk 3]
```diff
@@ -32,7 +29,7 @@ public SignUpResponse execute(SignUpRequest request) {
         String loginName = request.getLoginName();
         String password = request.getPassword();
         User createdUser = database.accessUserDatabase().save(new User(name, loginName, password, UserRole.CUSTOMER));
-        currentUserId.setValue(createdUser.getId());
+        request.getUserId().setValue(createdUser.getId());
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/actions/guest/SignUpService.java:32]: Request without a validator.

File: team_planet_express/src/main/java/core/services/actions/shared/SignInService.java [hunk 4]
```diff
@@ -28,7 +25,7 @@ public SignInResponse execute(SignInRequest request) {
             return new SignInResponse(errors);
         }
         User newUser = database.accessUserDatabase().findByLogin(request.getLoginName()).get();
-        currentUserId.setValue(newUser.getId());
+        request.getUserId().setValue(newUser.getId());
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/actions/shared/SignInService.java:28]: Id in validator?

File: team_planet_express/src/main/java/core/services/actions/shared/SignOutService.java [hunk 5]
```diff
@@ -5,21 +5,18 @@
...
+    public SignOutService(Database database) {
         this.database = database;
-        this.currentUserId = currentUserId;
     }
 
     public SignOutResponse execute(SignOutRequest request) {
         // TODO this dude has no cart
         User newUser = database.accessUserDatabase().save(new User("Guest", "", "", UserRole.GUEST));
-        currentUserId.setValue(newUser.getId());
+        request.getUserId().setValue(newUser.getId());
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/actions/shared/SignOutService.java:19]: No validator at all.

File: team_planet_express/src/test/java/console_ui/actions/guest/SignUpUIActionTest.java [hunk 6]
```diff
@@ -24,9 +25,10 @@ class SignUpUIActionTest {
     private final SignUpResponse mockSignUpResponse = mock(SignUpResponse.class);
     private final CoreError mockCoreError = mock(CoreError.class);
     private final User mockUser = mock(User.class);
+    private final MutableLong mockCurrentUserId = mock(MutableLong.class);
 
     private final SignUpUIAction action =
-            new SignUpUIAction(mockSignUpService, mockUserCommunication);
+            new SignUpUIAction(mockSignUpService, mockCurrentUserId,mockUserCommunication);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/console_ui/actions/guest/SignUpUIActionTest.java:31]: Ctrl+Alt+L

File: team_planet_express/src/test/java/core/services/actions/guest/SignUpServiceTest.java [hunk 7]
```diff
@@ -63,10 +65,12 @@ void shouldSaveValidRequest() {
     @Test
     void shouldUpdateCurrentUserId() {
         when(mockValidator.validate(mockRequest)).thenReturn(Collections.emptyList());
+
+        when(mockRequest.getUserId()).thenReturn(mockCurrentUserId);
         when(mockDatabase.accessUserDatabase()).thenReturn(mockUserDatabase);
         when(mockUserDatabase.save(any(User.class))).thenReturn(mockUser);
         service.execute(mockRequest);
-        verify(mockCurrentUserId).setValue(any(Long.class));
+        verify(mockRequest.getUserId()).setValue(any(Long.class));
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/guest/SignUpServiceTest.java:73]: Isn't verify(mockCurrentUserId).setValue(any(Long.class)); better?
It's easier to understand.

File: team_planet_express/src/test/java/core/services/actions/guest/SignUpServiceTest.java [hunk 8]
```diff
@@ -63,10 +65,12 @@ void shouldSaveValidRequest() {
     @Test
     void shouldUpdateCurrentUserId() {
         when(mockValidator.validate(mockRequest)).thenReturn(Collections.emptyList());
+
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/guest/SignUpServiceTest.java:68]: I also really do not like all those random newlines.

File: team_planet_express/src/test/java/core/services/actions/shared/SignInServiceTest.java [hunk 9]
```diff
@@ -63,9 +65,11 @@ void shouldUpdateCurrentUserId() {
         when(mockValidator.validate(mockRequest)).thenReturn(Collections.emptyList());
         when(mockRequest.getLoginName()).thenReturn("login");
         when(mockDatabase.accessUserDatabase()).thenReturn(mockUserDatabase);
+
+        when(mockRequest.getUserId()).thenReturn(mockCurrentUserId);
         when(mockUserDatabase.findByLogin("login")).thenReturn(Optional.of(mockUser));
         service.execute(mockRequest);
-        verify(mockCurrentUserId).setValue(any(Long.class));
+        verify(mockRequest.getUserId()).setValue(any(Long.class));
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/shared/SignInServiceTest.java:72]: Same as with SignUp.

File: team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java [hunk 10]
```diff
@@ -6,23 +6,23 @@
...
 
 public class SignOutServiceTest {
 
     private final Database fakeDatabase = new Database();
-    private final MutableLong mockCurrentUserId = mock(MutableLong.class);
     private final SignOutRequest mockRequest = mock(SignOutRequest.class);
 
-    private final SignOutService service = new SignOutService(fakeDatabase, mockCurrentUserId);
+    private final SignOutService service = new SignOutService(fakeDatabase);
+    private final MutableLong mockCurrentUserId = mock(MutableLong.class);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java:17]: Why did you move it?
It was happier together with its mock friend.

File: team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java [hunk 11]
```diff
@@ -6,23 +6,23 @@
...
 
     @Test
     void shouldChangeUser() {
         new FakeDatabaseInitializer(fakeDatabase).initialize();
+        when(mockRequest.getUserId()).thenReturn(mockCurrentUserId);
         service.execute(mockRequest);
         // Fake database creates 4 users..
-        verify(mockCurrentUserId).setValue(5L);
+        verify(mockRequest.getUserId()).setValue(5L);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java:25]: Same as with SignUp.

File: team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java [hunk 12]
```diff
@@ -6,23 +6,23 @@
...
         //verify(mockCurrentUserId).setValue(1L);
```
Comment (VsevolodShpagin) [inline team_planet_express/src/test/java/core/services/actions/shared/SignOutServiceTest.java:26]: Clear comments, when those are obviously useless.

File: team_planet_express/src/main/java/console_ui/actions/customer/AddItemToCartUIAction.java [hunk 13]
```diff
@@ -19,10 +20,13 @@ public class AddItemToCartUIAction extends UIAction {
     private final AddItemToCartService addItemToCartService;
     private final UserCommunication userCommunication;
 
-    public AddItemToCartUIAction(AddItemToCartService addItemToCartService, UserCommunication userCommunication) {
+    private final MutableLong currentUserId;
+
+    public AddItemToCartUIAction(AddItemToCartService addItemToCartService, MutableLong currentUserId, UserCommunication userCommunication) {
```
[the remark this answers is not in the document]
Comment (VsevolodShpagin) [reply team_planet_express/src/main/java/console_ui/actions/customer/AddItemToCartUIAction.java:24]: assignments in this.currentId etc. are still in different order, but i give up..

**Comment by arthurkov98** [comment]
I will rework validators. I need too much copy-paste to solve problem with validator id.

File: team_planet_express/src/main/java/core/services/validators/guest/SignUpValidator.java [hunk 14]
```diff
@@ -28,12 +31,19 @@ public SignUpValidator(Database database) {
...
         errors = new ArrayList<>();
+        validateUserId(request.getUserId()).ifPresent(errors::add);
         validateName(request.getName());
         validateLoginName(request.getLoginName());
         validatePassword(request.getPassword());
         return errors;
     }
 
+    private Optional<CoreError> validateUserId(MutableLong userId) {
+        return userId == null
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/validators/guest/SignUpValidator.java:42]: are we ok with it being blank or empty?

File: team_planet_express/src/main/java/core/services/validators/guest/SignUpValidator.java [hunk 15]
```diff
@@ -28,12 +31,19 @@ public SignUpValidator(Database database) {
...
     public List<CoreError> validate(SignUpRequest request) {
         errors = new ArrayList<>();
+        validateUserId(request.getUserId()).ifPresent(errors::add);
         validateName(request.getName());
         validateLoginName(request.getLoginName());
         validatePassword(request.getPassword());
         return errors;
     }
 
+    private Optional<CoreError> validateUserId(MutableLong userId) {
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/validators/guest/SignUpValidator.java:41]: is there a reason not to use validateIsPresent?

File: team_planet_express/src/main/java/core/services/validators/shared/SignInValidator.java [hunk 16]
```diff
@@ -34,6 +38,12 @@ public List<CoreError> validate(SignInRequest request) {
         return errors;
     }
 
+    private Optional<CoreError> validateUserId(MutableLong userId) {
+        return userId == null
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/validators/shared/SignInValidator.java:42]: is blank ok?

File: team_planet_express/src/main/java/core/services/validators/shared/SignInValidator.java [hunk 17]
```diff
@@ -34,6 +38,12 @@ public List<CoreError> validate(SignInRequest request) {
         return errors;
     }
 
+    private Optional<CoreError> validateUserId(MutableLong userId) {
```
Comment (VsevolodShpagin) [inline team_planet_express/src/main/java/core/services/validators/shared/SignInValidator.java:41]: why not validateIsPresent?
